pkg/app/piped/cloudprovider/kubernetes/state.go

```diff
@@ -108,6 +108,14 @@ func determineResourceHealth(key ResourceKey, obj *unstructured.Unstructured) (s
 		return determineClusterRoleHealth(obj)
 	case KindClusterRoleBinding:
 		return determineClusterRoleBindingHealth(obj)
+	case KindVirtualService:
+		return determineVirtualService(obj)
+	case KindDestinationRule:
+		return determineDestinationRule(obj)
+	case KindGateway:
+		return determineGateway(obj)
+	case KindServiceEntry:
+		return determineServiceEntry(obj)
```

Comment:
Since `IsKubernetesBuiltInResource` at L69 returns false due to lack in `builtInApiVersions` within `pkg/app/piped/cloudprovider/kubernetes/resourcekey.go`, it will never reach this point.